Nos/Views/CreateUsernameSheet.swift

```diff
@@ -0,0 +1,454 @@
+    private enum ClaimState {
+        /// There is no request in progress yet
+        case idle
+
+        /// The request is in progress
+        case claiming
+
+        /// The request finished successfully
+        case claimed
+
+        /// Something was wrong with the request
+        case failed(ClaimError)
+
+        var hasError: Bool {
+            error != nil
+        }
+
+        var error: ClaimError? {
+            switch self {
+            case .failed(let error):
+                return error
+            default:
+                return nil
+            }
+        }
+
+        var isIdle: Bool {
+            switch self {
+            case .idle:
+                return true
+            default:
+                return false
+            }
+        }
+    }
+
+    private var attributedUsername: AttributedString {
+        AttributedString(
+            username,
+            attributes: AttributeContainer([NSAttributedString.Key.foregroundColor: UIColor(Color.primaryTxt)])
+        ) + AttributedString(".nos.social")
+    }
+
+    private var showAlert: Binding<Bool> {
+        Binding {
+            claimState.hasError
+        } set: { _ in
+        }
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 20) {
+            Spacer(minLength: 40)
+            switch claimState {
+            case .idle, .claiming:
+                ProgressView()
+                    .tint(Color.accentColor)
+                    .scaleEffect(1.5)
+                    .frame(maxWidth: .infinity, maxHeight: .infinity)
+            case .failed(let error):
+                SwiftUI.Text(error.localizedDescription)
+                    .font(.clarity(.regular, textStyle: .callout))
+                    .foregroundStyle(Color.primaryTxt)
+            case .claimed:
+                PlainText(.localizable.excellentChoice).sheetTitle()
+                SwiftUI.Text(attributedUsername)
+                    .font(.clarity(.bold, textStyle: .title3))
+                    .foregroundStyle(Color.secondaryTxt)
+                SwiftUI.Text(
+                    LocalizedStringKey(
+                        String(localized: LocalizedStringResource.localizable.usernameClaimedNotice(username))
+                    )
+                )
+                .sheetDescription()
+
+                Spacer(minLength: 0)
+
+                Button(String(localized: LocalizedStringResource.localizable.done)) {
+                    isPresented = false
+                }
+                .buttonStyle(BigActionButtonStyle())
+            }
+            Spacer(minLength: 40)
+        }
+        .frame(maxWidth: .infinity)
+        .alert(isPresented: showAlert, error: claimState.error) {
+            Button {
+                isPresented = false
+            } label: {
+                SwiftUI.Text(.localizable.ok)
+            }
+        }
+        .task {
+            guard claimState.isIdle else {
```

Comment:
```suggestion
            guard case .idle = claimState else {
```

This would allow us to eliminate the `isIdle` computed property.

Reply:
I didn't know about that option, good to know.